youtube_dlc/downloader/hls.py

```diff
@@ -205,6 +205,75 @@ def is_ad_fragment_end(s):
                         break
                     i += 1
                     media_sequence += 1
+                elif line.startswith('#EXT-X-MAP'):
+                    if frag_index > 0:
+                        self.report_error(
+                            'initialization fragment found after media fragments, unable to download')
+                        return False
```

Comment:
If this happens, either `can_download` must correctly return `False` so that we can automatically hand over the download to `ffmpeg`,  or the download should be handed over manually (like what we do if pycrypto isnt found)